docs/examples/LapNIG_Hierarchy_1.py

```diff
@@ -0,0 +1,128 @@
+import numpy as np
+import scipy.stats as ss
+
+
+def like_lpdf(x, state):
+    mean = state[0]
+    scale = state[1]
+    return ss.laplace.logpdf(x, mean, scale)
+
+
+def initialize_state(hypers):
+    # print("----------- This is LapNIG_Hierarchy_1 -----------")
+    mean = hypers[0]
+    # var = hypers[1]
+    shape = hypers[2]
+    scale = hypers[3]
+    # mh_mean_var = hypers[4]
+    # mh_log_scale_var = hypers[5]
+    return [mean, scale / (shape + 1)]
+
+
+def initialize_hypers():
+    return [0, 10, 2, 1, 10, 1]
+
+
+def draw(state, hypers, rng):
+    mean = hypers[0]
+    var = hypers[1]
+    shape = hypers[2]
+    scale = hypers[3]
+    # mh_mean_var = hypers[4]
+    # mh_log_scale_var = hypers[5]
+    r1 = ss.norm.rvs(mean, np.sqrt(var), random_state=rng)
+    r2 = ss.invgamma.rvs(a=shape, loc=0, scale=scale, random_state=rng)
+
+    return [r1, r2]  # mean, scale
+
+
+def update_summary_statistics(x, add, sum_stats, state, cluster_data_values):
+    if not len(sum_stats):
+        sum_stats = [0, 0]
+    if add:
+        sum_stats[0] += abs(state[0] - x[0])
+        cluster_data_values = np.append(cluster_data_values, x)
+    else:
+        sum_stats[0] -= abs(state[0] - x[0])
+        ind = np.where(cluster_data_values == x)
+        cluster_data_values = np.delete(cluster_data_values, ind[0][0], ind[1][0])
+    return [sum_stats, cluster_data_values]
+
+
+def clear_summary_statistics(sum_stats):
```

Comment:
Do you think it's worth it to keep this method in python? Or it's better to move it back to C++?

Reply:
I would move as much code as possible to C++, to reduce the user's effort